backend/app/treasury/actions.py

```diff
@@ -21,22 +21,39 @@
 )
 
 
+async def make_transfers(treasury_address: str, asset: ERC20) -> list[Transfer]:
+    return (
+        await bitquery.get_eth_transfers(treasury_address)
+        if asset.token_symbol == "ETH"
+        else await get_token_transfers(treasury_address, asset.token_address)
+    )
+
+
+async def make_transfers_and_end_balance_for_treasury(
+    treasury: Treasury,
+) -> dict[str, tuple[list[Transfer], float]]:
+    transfers_and_end_balance: dict[str, tuple[list[Transfer], float]] = {}
+
+    for asset in treasury.assets:
+        transfers_for_asset = await make_transfers(treasury.address, asset)
+        if transfers_for_asset:
+            transfers_and_end_balance[asset.token_symbol] = (
+                transfers_for_asset,
+                asset.balance,
+            )
+
+    return transfers_and_end_balance
+
+
 async def make_transfers_balances_for_treasury(
     treasury: Treasury,
 ) -> BalancesAtTransfers:
     "Returns series of balances defined at times of assets transfers"
 
-    async def get_transfers(asset: ERC20) -> list[Transfer]:
-        return (
-            await bitquery.get_eth_transfers(treasury.address)
-            if asset.token_symbol == "ETH"
-            else await get_token_transfers(treasury.address, asset.token_address)
-        )
+    transfers_and_end_balance = await make_transfers_and_end_balance_for_treasury(
```

Comment:
If I read correctly, your changed the logic at the  `if transfers_for_asset` clause at line 39. That means that you are effectively making sure that the `BalancesAtTransfers` object doesn't contain any asset that has an empty transfers list.

What's the reason for this change? How is reducing the `BalancesAtTransfers` helping? Can't we go along well with some empty `Transfers` lists?

Reply:
I believe that's more of a job for `BalanceAtTransfers` itself: its constructor should be able to handle empty transfers (indeed, it seems totally reasonable that there is 0 transfers on a given period). That would make the code overall more robust. But I'm fine with this version for now.